Explorer/Assets/DCL/Infrastructure/ECS/StreamableLoading/AssetBundles/InitialSceneState/InitialSceneStateDescriptor.cs

```diff
@@ -0,0 +1,146 @@
+
+namespace ECS.StreamableLoading.AssetBundles.InitialSceneState
+{
+    public class InitialSceneStateDescriptor
+    {
+        public StreamableLoadingResult<AssetBundleData> AssetBundleData;
+        public AssetBundlePromise AssetBundlePromise;
+        public List<(string, GltfContainerAsset)> AssetsInstantiated;
+
+        private World GlobalWorld;
+        private string SceneID;
+        private AssetBundleManifestVersion AssetBundleManifestVersion;
+        private IGltfContainerAssetsCache assetsCache;
+
+        private bool AllAssetsInstantiated;
+        private bool IsSupportedByScene;
+
+        public bool IsDownloadedAndReady()
+        {
+            if (!IsSupportedByScene)
+                return true;
+
+            //The asset bundle failed to load for some reason...this is an escape route. The scene load will fail,
+            //abs seems to be corrupt
+            if (AssetBundleData.Exception != null)
+                return true;
+
+            //The asset bundle was destroyed at some point because of memory constrains. We got to nullify it and restart
+            if (AssetBundleData.IsInitialized && AssetBundleData.Asset!.AssetsDestroyed)
+            {
+                CreateEmptyAssetBundleData();
+                return false;
+            }
+
+            if (AssetBundleData.IsInitialized && AllAssetsInstantiated)
+                return true;
+
+            if (!AssetBundleData.IsInitialized && AssetBundlePromise == AssetBundlePromise.NULL)
+            {
+                //TOO (JUANI): Here we will use the sceneID to create the promise
+                AssetBundlePromise = AssetBundlePromise.Create(GlobalWorld,
+                    GetAssetBundleIntention.FromHash($"staticscene_{SceneID}{PlatformUtils.GetCurrentPlatform()}",
+                        assetBundleManifestVersion: AssetBundleManifestVersion,
+                        parentEntityID: SceneID),
+                    PartitionComponent.TOP_PRIORITY);
+            }
+
+            return false;
+        }
+
+        public void AddInstantiatedAsset(string hash, GltfContainerAsset asset)
+        {
+            AssetsInstantiated.Add((hash, asset));
+            AllAssetsInstantiated = AssetsInstantiated.Count == AssetBundleData.Asset.InitialSceneStateMetadata.Value.assetHash.Count;
+        }
+
+        public static InitialSceneStateDescriptor CreateUnsupported(string sceneID)
+        {
+            InitialSceneStateDescriptor unsuportedStaticSceneAB = new InitialSceneStateDescriptor();
+            unsuportedStaticSceneAB.AssetBundleData = new StreamableLoadingResult<AssetBundleData>(ReportCategory.ASSET_BUNDLES, new Exception($"Static Scene Asset Bundle not suported for {sceneID}"));
+            unsuportedStaticSceneAB.AssetsInstantiated = new List<(string,GltfContainerAsset)>();
```

Comment:
Can you get away with `IReadOnlyList` and `Array.Empty` to prevent creation of a new empty list?